service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java

```diff
@@ -247,6 +256,59 @@ public void listWorkspaces_requesterIsDiscoverer_requestMinHighestRoleNotSet_ret
     assertTrue(listedWorkspaces.isEmpty());
   }
 
+  @Test
+  public void grantRole_logsAnActivity() throws Exception {
+    mockMvcUtils.grantRole(
+        userAccessUtils.defaultUserAuthRequest(),
+        workspace.getId(),
+        WsmIamRole.DISCOVERER,
+        userAccessUtils.getSecondUserEmail());
+
+    ActivityLogChangeDetails changeDetails = getLastChangeDetails(workspace.getId());
```

Comment:
Same for test file and WorkspaceApiControllerTest, create helper methods. One of the READMEs refers to these tests, so people may copy-paste from here. In a previous PR, I went through these files and moved stuff to helper methods.

Reply:
moved it to MockMvcUtil. mirror assertClonedResourceMetadata()